### ISSUE TITLE
Errors seen after hosted rancher is upgraded to 2.6 and is managing a 2.5 rancher setup

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#34961]
**Rancher Server Setup**
- Rancher version: 2.6.1-rc5 hosted managing 2.5.9 rancher
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): k3s `1.20.11+k3s1`
- Proxy/Cert Details: Rancher generated

**Information about the Cluster**
- Kubernetes version: Downstream - 1.20.11+k3s1 rancher. K3s
- Cluster Type (Local/Downstream): Downstream - Tenant rancher - 2.5.9 rancher installed

**Describe the bug**
Errors seen after hosted rancher is upgraded to 2.6 and is managing a 2.5 rancher setup

**To Reproduce**

- On 2.5.9 rancher, deploy a downstream tenant rancher server
- Upgrade hosted rancher to 2.5.1-rc5
- Error seen in rancher pod:

```
2021/09/29 05:22:44 [ERROR] error syncing 'local/high-number-of-leader-changes': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/db-over-size': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/kube-components-alert': handler cluster-alert-group-deployer: can't find system project, handler cluster-alert-group-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/etcd-alert': handler cluster-alert-group-deployer: can't find system project, handler cluster-alert-group-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'cattle-global-nt': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'cattle-global-data': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/event-alert': handler cluster-alert-group-deployer: can't find system project, handler cluster-alert-group-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/scheduler-system-service': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/high-cpu-load': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/cluster-scan-manual-all': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/cluster-scan-scheduled-all': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/deployment-event-alert': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/no-leader': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/controllermanager-system-service': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/cluster-scan-scheduled--failure-only': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/node-disk-running-full': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/high-memmory': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/etcd-system-service': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/cluster-scan-manual-failure-only': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'default': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-whr9p" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-whr9p" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'kube-node-lease': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'cert-manager': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'cattle-system': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'kube-public': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'kube-system': handler resourceQuotaSyncController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "c-z7mx5/p-8v7c9" not found, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/cluster-scan-alert': handler cluster-alert-group-deployer: can't find system project, handler cluster-alert-group-controller: can't find system project, requeuing
2021/09/29 05:22:44 [ERROR] error syncing 'local/node-alert': handler cluster-alert-group-deployer: can't find system project, handler cluster-alert-group-controller: can't find system project, requeuing
```
- Downstream cluster in tenant rancher (EC2 rke1 cluster) is not accessible from rancher. No errors seen in pods in the local cluster
-   Upgrade tenant to 2.6.1-rc5
- Downstream cluster in tenant rancher is available.
- No such errors seen in tenant cluster rancher pods.

**Comment by aiyengar2** [comment]
@sowmyav27 I was able to reproduce the class of errors that look like this:
```log
2021/09/29 05:22:44 [ERROR] error syncing 'NAMESPACE': handler resourceQuotaSyncController: projects.management.cattle.io "CLUSTER_ID/PROJECT_NAME" not found, handler resourceQuotaUsedLimitController: projects.management.cattle.io "CLUSTER_ID/PROJECT_NAME" not found, requeuing
```

The root cause of this issue is that the downstream Rancher tenant tries to reconcile ResourceQuotas in a namespace deployed by the super Rancher (and tracked in one of the super Rancher's projects, which is why that project results in `not found`).

The fix for this should be simply back-porting https://github.com/rancher/rancher/pull/34938, which tells Rancher to ignore calculating resource quotas or syncing app state for projects that it is unaware of.

---

For the second class of errors:

```log
2021/09/29 05:22:44 [ERROR] error syncing 'local/CLUSTER-ALERT-CR': handler cluster-alert-rule-deployer: can't find system project, handler cluster-alert-rule-controller: can't find system project, requeuing
```

The issue here seems to indicate that the System Project is not found in the downstream Rancher; I was unable to reproduce this and it is possible this is just an ephemeral error that happens as the system project is created. Can you reproduce?

---

Pending what is required to fix the second issue, I think this issue can be moved out of `2.6.1` in favor of a `2.5.10` milestone, since the only fixes I anticipate now are to Rancher 2.5.

**Comment by sowmyav27** [comment]
**On 2.5-head commit id: `ce52c78`**

Validated the issue and fix works as expected

**Steps:**
- Deployed Hosted rancher on 2.5.0
- Deployed downstream tenant1 - 2.5.9
- Deployed downstream tenant2 - 2.5.9
- Upgraded tenant1 to 2.5-head
- Upgraded Hosted rancher to 2.6.1-rc11
- Noticed the errors mentioned in the original issue in the tenant rancher pods running on rancher 2.5.9 but NOT on tenant running on rancher 2.5-head
- Upgraded tenant2 to 2.5-head. No errors seen post upgrade.
